### PR TITLE
Send 100 Continue when ready to receive topic downstream

### PR DESCRIPTION
**Pull request by patuwwy**
**What?**  

### :large_blue_diamond: Disable auto sending HTTP 100 Continue
NodeJs auto sends `HTTP 100 Continue` when it is not strictly handled on server. 
We handle it in other way and send when needed so it is sent twice.

_https://nodejs.org/api/http.html#event-checkcontinue_
_https://github.com/nodejs/node/blob/v18.x-staging/lib/_http_server.js#L1110_

### :large_blue_diamond: Topic downstream queue update
- Require ```expect: 100-continue``` header in topic downstream endpoint.
- Send `HTTP 100 Continue` when request can be processed.

### :large_blue_diamond: Refactoring
Move ```TopicRouter``` / ```ServiceDiscovery``` types to ```@scramjet/types```

--- 

:heavy_exclamation_mark: User should wait for ```HTTP 100 Continue``` and then start to upload data. 

:heavy_exclamation_mark: ```@scramjet/cli``` / ```@scramjet/api-client``` CAN'T listen for ```100 Continue``` as we use ```node-fetch``` which doesn't support `continue` request's event. Data will be sent immediately. 

If data are sent with no wait for `100 Continue` some bytes can be lost when request is interrupted before we started handling these data.

---

**Review checks:**

These aspects need to be checked by the reviewer:

- [ ] Verify and confirm operation (please post a screenshot) 
- [x] All STH tests pass
- [ ] All [Scramjet Cloud Platform](https://docs.scramjet.org/platform) tests pass
- [ ] Documentation is updated or no changes

### CODE REVIEW

**Review by alicja-gruzdz — APPROVED** [review]

File: packages/host/src/lib/serviceDiscovery/topicRouter.ts [hunk 1]
```diff
@@ -116,7 +119,11 @@ class TopicRouter {
             };
         }
 
-        topic.acceptPipe(req);
+        Object.assign(
+            req, { writeContinue: () => { res.writeContinue(); } }
+        );
```
[the remark this answers is not in the document]
Comment (MichalCz) [reply packages/host/src/lib/serviceDiscovery/topicRouter.ts:124]: Error handling needed (check if expect: continue is there - so maybe if not, in server just add a `const noop = () => 0`).

**Review by MichalCz — APPROVED** [review]

File: packages/host/src/lib/serviceDiscovery/topicRouter.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (patuwwy) [reply packages/host/src/lib/serviceDiscovery/topicRouter.ts]: https://github.com/scramjetorg/transform-hub/pull/1010/files#diff-4c21861bc607c58b6693eb12e08915fc1fe6a00c154e7a8e541acf3087037446R104-R112

**Comment by a-tylenda** [comment]
verified

image

**Comment by a-tylenda** [comment]
image

**Pull request merged**